src/platform/i386/timer.c

```diff
@@ -7,37 +7,249 @@
 #include "io.h"
 #include "kernel.h"
 
-#define PIT_A       0x40
-#define PIT_B       0x41
-#define PIT_C       0x42
-#define PIT_CONTROL 0x43
-#define PIT_SET     0x36
-#define PIT_MASK    0xFF
-#define PIT_SCALE   1193180
+#define rdtscll(val) __asm__ __volatile__("rdtsc" : "=A" (val))
 
-/* FIXME: per-thread */
-static struct thread *timer_thread = NULL;
+typedef struct {
```

Comment:
I'm not sure we want to completely get rid of the PIT.  We want a performance comparison, and I think the PIT is faster to write to.  WE won't use it, but we likely want the comparison.